### PR TITLE
Add DeleteObjectSetFromDeviceRpc.

### PR DESCRIPTION
**Pull request by shinnya**
I've added a new rpc for deleting objects from a device.

### CODE REVIEW

File: src/entity/object.rs [hunk 1]
```diff
@@ -10,6 +12,18 @@ pub type ObjectId = String;
 )]
 pub struct ObjectVersion(pub u64);
 
+impl FromStr for ObjectVersion {
```
Comment (shinnya) [inline src/entity/object.rs:15]: Used when parsing a command-line option.

File: src/client/frugalos.rs [hunk 2]
```diff
@@ -179,6 +181,25 @@ impl Client {
         )
     }
 
+    /// Executes `DeleteObjectSetFromDeviceRpc`.
+    pub fn delete_from_device_by_object_ids(
+        &self,
+        bucket_id: BucketId,
+        device_id: DeviceId,
+        object_ids: BTreeSet<ObjectId>,
+    ) -> impl Future<Item = (), Error = Error> {
```
Comment (shinnya) [inline src/client/frugalos.rs:190]: `DeviceId` is required here because we delete objects from a specific device.

File: src/client/frugalos.rs [hunk 3]
```diff
@@ -179,6 +181,25 @@ impl Client {
         )
     }
 
+    /// Executes `DeleteObjectSetFromDeviceRpc`.
+    pub fn delete_from_device_by_object_ids(
+        &self,
+        bucket_id: BucketId,
+        device_id: DeviceId,
+        object_ids: BTreeSet<ObjectId>,
```
Comment (shinnya) [inline src/client/frugalos.rs:189]: Use `BTreeSet` because we'd like to remove duplicated objects for efficiency.

File: src/entity/object.rs [hunk 4]
```diff
@@ -10,6 +12,18 @@ pub type ObjectId = String;
...
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        track!(
+            s.parse::<u64>()
+                .map(|n| ObjectVersion(n))
```
Comment (sile) [inline src/entity/object.rs]: `clippy` warns this line ( https://travis-ci.org/frugalos/libfrugalos/jobs/457307395#L576 ).

File: src/entity/device.rs [hunk 5]
```diff
@@ -11,6 +11,9 @@ use {Error, ErrorKind, Result};
 /// デバイスのID。
 pub type DeviceId = String;
 
+/// the sequence number of devices, which is increased monotonically.
```
Comment (sile) [inline src/entity/device.rs]: A sentence should start with an upper case character.
See also: https://doc.rust-lang.org/1.0.0/style/style/comments.html#sentence-structure

File: src/schema/frugalos.rs [hunk 6]
```diff
@@ -171,6 +173,22 @@ impl Call for DeleteObjectsByPrefixRpc {
     type ResEncoder = BincodeEncoder<Self::Res>;
 }
 
+/// A RPC for deleting objects physically.
```
Comment (sile) [inline src/schema/frugalos.rs]: Perhaps, "An" is correct for "RPC".

File: src/schema/frugalos.rs [hunk 7]
```diff
@@ -229,6 +247,19 @@ pub struct SegmentRequest {
     pub segment: u16,
 }
 
+/// This struct represents how to delete objects from a device at once.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DeleteObjectSetFromDeviceRequest {
+    /// a bucket which owns the objects
```
Comment (sile) [inline src/schema/frugalos.rs]: See: https://doc.rust-lang.org/1.0.0/style/style/comments.html#sentence-structure

**Review by sile — CHANGES_REQUESTED** [review]

File: src/entity/object.rs (hunk 4 above)
Comment (shinnya) [reply src/entity/object.rs]: fixed

**Review by sile — APPROVED** [review]
LGTM

**Pull request merged**